pymeasure/instruments/hp/hpsystempsu.py

```diff
@@ -0,0 +1,330 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import ctypes
+import logging
+
+from enum import Enum
+from pymeasure.instruments.hp.hplegacyinstrument import HPLegacyInstrument, StatusBitsBase
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+c_uint8 = ctypes.c_uint8
+
+
+class Status(StatusBitsBase):
+    """
+    Support-Class with the bit assignments for the 2 status byte(12bits used) of the HP6632A
+    """
+
+    _fields_ = [
+        # Byte 1: Function, Range and Number of Digits
+        ("Error_pending", c_uint8, 1),  # bit 7
+        ("Overcurrent", c_uint8, 1),  # bit 6
+        ("not_assigned", c_uint8, 1),  # bit 5
+        ("Overtempeature", c_uint8, 1),  # bit 4
+        ("Overvoltage", c_uint8, 1),  # bit 3
+        ("Unregulated", c_uint8, 1),  # bit 2
+        ("CCpos", c_uint8, 1),  # bit 1
+        ("CV", c_uint8, 1),  # bit 0
+
+        # Byte 2: Status Bits
+        ("not_assigned", c_uint8, 4),  # bits 16..12
+        ("NORM", c_uint8, 1),  # bit 11
+        ("FAST", c_uint8, 1),  # bit 10
+        ("CCneg", c_uint8, 1),  # bit 9
+        ("Inhibit_active", c_uint8, 1),  # bit 8
+
+    ]
+
+
+limits = {
+    "HP6632A": {"Volt_lim": 20.475, "OVP_lim": 22.0, "Cur_lim": 5.118},
+    "HP6633A": {"Volt_lim": 51.118, "OVP_lim": 55.0, "Cur_lim": 2.0475},
+    "HP6634A": {"Volt_lim": 102.38, "OVP_lim": 110.0, "Cur_lim": 1.0238}}
+
+
+class HP6632A(HPLegacyInstrument):
+    """ Represents the Hewlett Packard 6632A system power supply
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+    status_desc = Status
+
+    def __init__(self, resourceName, **kwargs):
+        kwargs.setdefault('read_termination', '\r\n')
+        kwargs.setdefault('send_end', True)
+        super().__init__(
+            resourceName,
+            "Hewlett-Packard HP6632A",
+            status_bitfield=self.status_desc,
+            **kwargs,
+        )
+
+    class ERRORS(Enum):
+        """
+        Enum class for error messages
+
+        """
+        NO_ERR = 0
+        EEPROM = 1
+        PON_2ND = 2
+        DCPON_2MD = 4
+        NORELAY = 5
+        NOTHING_SAY = 8
+        HEADER_EXPECTED = 10
+        UNRECOGNISED_HEADER = 11
+        NUMBER_EXPECTED = 20
+        NUMBER_SYNTAX = 21
+        OUT_OF_RANGE = 22
+        COMMA_EXPECTED = 30
+        TERM_EXPECTED = 31
+        PARAM_OUT = 41
+        V_PGM_ERR = 42
+        I_PGM_ERR = 43
+        OV_PGM_ERR = 44
+        DLY_PGM_ERR = 45
+        MASK_PGM_ERR = 46
+        MULT_CSAVE_ERR = 50
+        EEPROM_CJLSUM_ERR = 51
+        CALMODE_DISABLED = 52
+        CAL_CHNL_ERR = 53
+        CAL_FS_ERR = 54
+        CAL_OFFSET = 55
+        CAL_JMP_ERR = 59
+
+    class ST_ERRORS(Enum):
```

Comment:
This class is not used anywhere. From comment it seems that it should be used in method `check_selftest_errors`

Reply:
Yes you are correct, that was a probably a copy& paste with some missing edit.
changed the call in `check_selftest_errors()` to use the `ST_ERRORS` class